dot/state/epoch.go

```diff
@@ -884,12 +1006,62 @@ func (s *EpochState) FinalizeBABENextEpochData(finalizedHeader *types.Header) er
 	for e := range s.nextEpochData {
 		if e <= nextEpoch {
 			delete(s.nextEpochData, e)
+			// remove the epoch data from the database
+			if err = s.deleteEpochDataFromDisk(e); err != nil {
+				return fmt.Errorf("cannot delete next epoch data from the database: %w", err)
+			}
 		}
 	}
 
 	return nil
 }
 
+func (s *EpochState) deleteEpochDataFromDisk(epoch uint64) error {
+	configKeysToDelete, err := getDataKeysFromDisk[types.NextEpochData](s.db, nextEpochDataPrefix, epoch)
+	if err != nil {
+		return fmt.Errorf("cannot get next config data keys from disk: %w", err)
+	}
+
+	for _, key := range configKeysToDelete {
+		err = s.db.NewBatch().Del([]byte(key))
+		if err != nil {
+			return fmt.Errorf("cannot delete next config data from the database: %w", err)
+		}
+	}
```

Comment:
You're not flushing the batcher, so the deletion operation is not being performed
```suggestion
        deletionBatch := s.db.NewBatch()
	for _, key := range configKeysToDelete {
		err = deletionBatch.Del([]byte(key))
		if err != nil {
			return fmt.Errorf("cannot delete next config data from the database: %w", err)
		}
	}
	deletionBatch.Flush()
```